backend/src/routes/images.ts

```diff
@@ -0,0 +1,56 @@
+const express = require('express');
+const router = express.Router();
+import {Request, Response} from 'express';
+import Image from '../models/ImageModel';
+
+router.post('/create', async function(req: Request, res: Response) {   
+    const colors: string[] = req.body.colors;
+
+    const isStrings = (arr: string[]) => arr.every(i => typeof i === 'string')
+    console.log(isStrings(colors));
+    
+
+    if (isStrings(colors) && colors.length > 1 && colors.length < 5) {
+        type squareType = {
+            squares: {[key: number]: string}          
+          };
+
+        const image: squareType = {
+            squares: {}
+        }
+
+        for (let i:number = 0; i <225; i++) {
+            const colorAmount = req.body.colors.length;
+            const colorIndex = Math.floor(Math.random()*colorAmount);         
+            image.squares[i] = req.body.colors[colorIndex]            
+        }         
+        
+    const newImage = new Image(image);
```

Comment:
Här skulle ni kunna göra nånting sånt här

`const newImage = await Image.create(image)
if (newImage) {
  res.status(201)
} else {
  res.status(500)
  throw new Error('fel')
}
`

Reply:
Vi tycker att den oneliner-koden blir lite för lång och att try-catch-koden lite mer överskådlig samt att vi skickar in lite mer info om ev fel i svaret. Dock tog vi "await Image.create(image)" för att vänta på mongoose.